### DISCUSSION TITLE
`with-items` using `jinja`

### DISCUSSION THREAD

**Discussion by jpavlav** [StackStorm/st2#5973] [Q&A] [answered]
Hey folks!

Ran into an interesting (what I thought was interesting) quandary when trying to use `with-items` with `jinja`. Specifically, the third example in this documentation:

https://docs.stackstorm.com/orquesta/languages/orquesta.html#with-items-model

shows how to `name` the items iterated over when designating `with`. However, it only shows how to do this in `yaql`. I'm curious if someone can provide an example of doing the same with `jinja`. I tried a bunch of different permutations, but never got it to work. Any help would be much appreciated.

TLDR; How to name items when using `with-items` and `jinja` templating?

**Comment by nzlosh** [comment, marked as answer]
You might be running into this issue https://github.com/StackStorm/orquesta/issues/247

**Reply by jpavlav** [reply]
Ah interesting, though, the responses on this issue appear to actually answer the question I had, how to construct `with-items` using a `name` for the `item`:

```yaml
tasks:
  echo_input:
    with: name in {{ ctx().names }}
    action: core.echo
    input:
      message: "Hello {{ item(name) }}"
```

Have yet to test, but the `st2 workflow inspect` command isn't complaining about it, so I suppose I'll find out.

**Reply by jpavlav** [reply]
Okay, looks like it does work, just needed to throw some quotes around `name` when it was consumed by the `item` function:

```yaml
tasks:
  echo_input:
    with: name in {{ ctx().names }}
    action: core.echo
    input:
      message: "Hello {{ item('name') }}"
```

Thanks for the help!
